StyleCop.Analyzers/StyleCopTester/Program.cs

```diff
@@ -24,6 +25,35 @@ internal static class Program
     {
         private static void Main(string[] args)
         {
+            CancellationTokenSource cts = new CancellationTokenSource();
+            Console.CancelKeyPress +=
+                (sender, e) =>
+                {
+                    e.Cancel = true;
+                    cts.Cancel();
+                };
+
+            SynchronizationContext previousContext = SynchronizationContext.Current;
+            try
+            {
+                var context = new DispatcherSynchronizationContext();
```

Comment:
If we get to a point where the project runs on Mono, this can be replaced with our own `SynchronizationContext` implementation without breaking the rest of the changes. We could probably even keep the same names for the types and just declare them as nested types in the file.